internal/impl/mongodb/cdc/integration_test.go

```diff
@@ -791,3 +808,79 @@ mongodb_cdc:
 	wait()
 	require.JSONEq(t, `[{"_id":1,"data":"hello"}, {"_id":2,"data":"hello"}, {"_id":3,"data":"hello"}]`, output.MessagesJSON(t))
 }
+
+func TestIntegrationMongoCDCWithSnapshotShardedCluster(t *testing.T) {
+	integration.CheckSkipExact(t)
+	// You can setup a sharded cluster with https://github.com/pkdone/sharded-mongodb-docker
```

Comment:
I'd consider adding it as a submodule here and running compose in the test.
Dockertest does not support compose directly.